### ISSUE TITLE
Conditional allocations shouldn't fail if cond=false, size=0 -> nullptr result

### ISSUE THREAD

**Issue by steven-johnson** [halide/Halide#7255]
Allocations can be conditional; if the condition evaluates to false, we end up calling `halide_malloc(0)`. Since it's legal via spec for `malloc(0)` to return nullptr, we need to be cautious here: if we are compiling with assertions enabled, *and* have a malloc() (etc) implementation that returns nullptr for alloc(0), we need to skip the assertion check, since we know the result won't be used... but the assertion would fail because the (ignored) result is null.

**Comment by abadams** [comment]
This may be an issue unique to the C backend, if that's where you observed this. I see code that handles this in CodeGen_Posix.cpp:307

**Comment by steven-johnson** [comment]
Yep. That logic looks simpler than what I have for a fix, I'll try replicating that instead. Fix on the way either way.

**Comment by steven-johnson** [comment]
FYI: via inserting a crash at Codegen_Posix.cpp:298 and rebuilding the universe, I can't find a single AOT case in our tests that actually triggers the `!is_const_one()` case. (Gonna test JIT shortly)

**Comment by abadams** [comment]
> [!NOTE]
> **Automated triage:** this issue was flagged by Claude as likely safe to close, for the reason below. The reasoning was checked against the current source, and a second pass tried and failed to refute it, but no human verified this specific text. **If the reason is wrong, please reopen the issue and say so** — some fraction of these are expected to be wrong.

**Reason (CLOSE_FIXED):** The C++ backend now emits the same relaxations the LLVM backend had: the out-of-memory assertion is skipped entirely for a statically-false condition, and otherwise allows a null result when the size is zero or the condition is false.

**Evidence:** Fixed by commit 04bb98616 "Conditional allocations shouldn't fail for size=0 in C++ backend (#7255) (#7256)"; current code at src/CodeGen_C.cpp:2419-2436 emits `(op_name != nullptr) || (size == 0) || (!condition)`, matching src/CodeGen_CPU.cpp:283-294.

Fixed by 04bb98616; the C++ backend now emits `(ptr != nullptr) || (size == 0) || (!condition)` for the allocation check (src/CodeGen_C.cpp:2419-2436), matching the LLVM backend. Closing.
